src/header/mod.rs

```diff
@@ -385,6 +385,19 @@ impl ProtectedHeader {
         ))
     }
 
+    pub fn to_be_authenticated(self) -> Result<Value> {
+        let result = self.cbor_bstr()?;
+        // protected header might have been encoded as a zero length map, only containing
+        // the byte 0xA0 (see RFC 9052, Section 3).
+        // However, this byte should not be included during authentication (RFC 9052, Section 4.4,
+        // 5.3 and 6.3), so we need to return an empty byte string here instead.
```

Comment:
When would `ProtectedHeader::cbor_bstr()` ever return an (encoded) empty map?  

I _think_ the `is_empty()` check would mean that this could only happen if the original input (in `ProtectedHeader::original_data`) had the non-minimal encoding, and that object then got re-encoded using the original data.  Or is there another way it can happen?

Reply:
That should be the only way that this can happen, yes.

For actually re-encoding an existing COSE object this should not be an issue (hence why `cbor_bstr()` is unchanged).

However, as far as I understand the COSE specification for generating the structures that are provided to the signature/encryption/MAC-function (e.g. `Sig_structure`), this non-minimal encoding needs to be [explicitly replaced with a zero-length byte string](https://datatracker.ietf.org/doc/html/rfc9052#section-4.4-2.2) if no attributes are set, regardless of whether the actual encoding of the protected header is minimal or not.

The aforementioned examples/test cases explicitly check for this behavior (e.g. [sign1-tests/sign-pass-01](https://github.com/cose-wg/Examples/blob/master/sign1-tests/sign-pass-01.json) and [sign-tests/sign-pass-01](https://github.com/cose-wg/Examples/blob/master/sign-tests/sign-pass-01.json)), and in my own tests verification of these (supposedly valid) COSE structures only works with the changes made in this PR.